pkg/counter/cache.go

```diff
@@ -117,6 +122,29 @@ func DecrementUserAccountCount(logger logr.Logger, clusterName string) {
 	})
 }
 
+// IncrementSpaceCount increments the number of Space's for the given member cluster in the cached counter
+func IncrementSpaceCount(logger logr.Logger, clusterName string) {
+	write(func() {
+		cachedCounts.SpacesPerClusterCounts[clusterName]++
+		logger.Info("incremented SpacesPerClusterCounts", "clusterName", clusterName, "value", cachedCounts.SpacesPerClusterCounts[clusterName])
+		metrics.SpaceGaugeVec.WithLabelValues(clusterName).Set(float64(cachedCounts.SpacesPerClusterCounts[clusterName]))
+	})
+}
+
+// DecrementSpaceCount decreases the number of Spaces for the given member cluster in the cached counter
+func DecrementSpaceCount(logger logr.Logger, clusterName string) {
+	write(func() {
+		if cachedCounts.SpacesPerClusterCounts[clusterName] != 0 || !cachedCounts.initialized { // counter can be decreased even if its current value is `0`, but only if the cache has not been initialized yet
+			cachedCounts.SpacesPerClusterCounts[clusterName]--
+			metrics.SpaceGaugeVec.WithLabelValues(clusterName).Set(float64(cachedCounts.SpacesPerClusterCounts[clusterName]))
+			logger.Info("decremented Spaces count", "value", cachedCounts.SpacesPerClusterCounts[clusterName])
+		} else {
+			logger.Error(fmt.Errorf("the count of Spaces is zero"),
+				"unable to decrement the number of Spaces for the given cluster", "cluster", clusterName)
+		}
+	})
+}
+
```

Comment:
this looks good, but you need to update also the functions:
* `Synchronize`
* `initialize`
* `initializeFromResources`
* `initializeFromToolchainStatus`

so the values between ToolchainStatus, Gauge, and the cache are in sync and properly initialized

Reply:
Yes, this is still very WIP, I spent some time on getting the build working and now trying to work on finishing this and adding tests. Thanks!

Reply:
ok, I see.
If you want to open a PR that is still WIP and you don't expect any review to be made, then please open it as a draft. 
I noticed the WIP prefix, but I thought that you expect some feedback/review since it's opened as a normal PR.

Reply:
This was very useful pointing me in the right direction for completing the changes!

Ok so Draft is something that people will not review , understood!  
Even if WIP prefix has the same effect I think (applies the `do-not-merge/work-in-progress` label).

I usually prefer keeping PR in WIP so that:

1. everyone can review and stay in synch 
2. (in this case) it triggers the GitHub actions and code gets tested ...